tests/baselines/reference/narrowingOfQualifiedNames.errors.txt

```diff
@@ -0,0 +1,70 @@
+tests/cases/compiler/narrowingOfQualifiedNames.ts(35,33): error TS2532: Object is possibly 'undefined'.
+tests/cases/compiler/narrowingOfQualifiedNames.ts(37,37): error TS2532: Object is possibly 'undefined'.
+tests/cases/compiler/narrowingOfQualifiedNames.ts(40,41): error TS2532: Object is possibly 'undefined'.
+tests/cases/compiler/narrowingOfQualifiedNames.ts(40,41): error TS2532: Object is possibly 'undefined'.
+tests/cases/compiler/narrowingOfQualifiedNames.ts(42,45): error TS2532: Object is possibly 'undefined'.
+tests/cases/compiler/narrowingOfQualifiedNames.ts(42,45): error TS2532: Object is possibly 'undefined'.
+
+
+==== tests/cases/compiler/narrowingOfQualifiedNames.ts (6 errors) ====
+    // Repro from #43411
+    
+    interface IProperties {
+        foo?: {
+            aaa: string
+            bbb: string
+        }
+    }
+    
+    function init(properties: IProperties) {
+        if (properties.foo) {
+            type FooOK = typeof properties.foo;
+            properties.foo; // type is { aaa: string; bbb: string; }
+            for (const x of [1, 2, 3]) {
+              properties.foo; // type is { aaa: string; bbb: string; }
+              type FooOrUndefined = typeof properties.foo; //type is { aaa: string; bbb: string; } | undefined
+            }
+        }
+    }
+    
+    interface DeepOptional {
+        a?: {
+            b?: {
+                c?: string
+            }
+        }
+    }
+    
+    function init2(foo: DeepOptional) {
+        if (foo.a) {
+            for(const _ of [1]) {
+                type A = typeof foo.a;
+    
+                if (foo.a.b) {
+                    type B = typeof foo.a.b;
+                                    ~~~~~
+!!! error TS2532: Object is possibly 'undefined'.

```

Comment:
IMO, this is a bug. Should it be fixed in this PR?

Reply:
Yes, I'm pretty sure that possibly-undefined errors also use control flow, so this is a sign that something else needs to be fixed.

Reply:
```diff
                case SyntaxKind.QualifiedName:
-                   if (currentFlow && parent.kind === SyntaxKind.TypeQuery) {
+                   if (currentFlow && isPartOfTypeQuery(node)) {
                        node.flowNode = currentFlow;
                    }
```

binding children of `QualifiedName` fixed it.

https://github.com/microsoft/TypeScript/pull/43592/files#diff-9ad14a2e2e66533cd0c3f02ecf01f7e6bc9d60cdf2e125df2a14ce0713222898R2521